core/src/View/MDSpan/Kokkos_MDSpan_Accessor.hpp

```diff
@@ -169,6 +170,61 @@ struct SpaceAwareAccessor<AnonymousSpace, NestedAccessor> {
   friend struct SpaceAwareAccessor;
 };
 
+// Like atomic_accessor_relaxed proposed for ISO C++26 but with
+// defaulted memory scope - similar to how desul's AtomicRef has a memory scope
+template <class ElementType, class MemoryScope = desul::MemoryScopeDevice>
+struct AtomicAccessorRelaxed {
+  using element_type = ElementType;
+  using reference =
+      desul::AtomicRef<ElementType, desul::MemoryOrderRelaxed, MemoryScope>;
+  using data_handle_type = ElementType*;
+  using offset_policy    = AtomicAccessorRelaxed;
+
+  KOKKOS_DEFAULTED_FUNCTION
+  AtomicAccessorRelaxed() = default;
+
+  // TODO:  Kokkos allows atomic views of const element types right now,
+  //        but for the future we might to deprecated it and do this:
+#if 0
+  KOKKOS_FUNCTION
+  AtomicAccessorRelaxed(const default_accessor<element_type>&) {}
+
+  static_assert(
+      std::is_same_v<std::remove_cv_t<element_type>, element_type>,
+      "AtomicAccessorRelaxed can only be used for non-const element types");
+#endif
```

Comment:
```suggestion
```
Since we now have an issue we should remove dead code https://github.com/kokkos/kokkos/issues/7115

Reply:
Done